core-rust/state-manager/src/store/column_families.rs

```diff
@@ -0,0 +1,471 @@
+    type Key = ScenarioSequenceNumber;
+    type Value = ExecutedScenario;
+
+    // Note: a legacy name is still used here, even though we now have scenarios run outside Genesis
+    const VERSIONED_NAME: &'static str = "executed_genesis_scenarios";
+    type KeyCodec = ScenarioSequenceNumberDbCodec;
+    type VersionedValue = VersionedExecutedScenario;
+}
+
+/// Progress of the GC process pruning the [`LedgerProofsCf`].
+/// Schema: `[]` -> `scrypto_encode(VersionedLedgerProofsGcProgress)`
+/// Note: This is a single-entry table (i.e. the empty key is the only allowed key).
+pub struct LedgerProofsGcProgressCf;
+impl VersionedCf for LedgerProofsGcProgressCf {
+    type Key = ();
+    type Value = LedgerProofsGcProgress;
+
+    const VERSIONED_NAME: &'static str = "ledger_proofs_gc_progress";
+    type KeyCodec = UnitDbCodec;
+    type VersionedValue = VersionedLedgerProofsGcProgress;
+}
+
+/// Node IDs and blueprints of all entities, indexed by their type and creation order.
+/// Schema: `[EntityType as u8, StateVersion.to_be_bytes(), (index_within_txn as u32).to_be_bytes()].concat()` -> `scrypto_encode(VersionedEntityBlueprintId)`
+pub struct TypeAndCreationIndexedEntitiesCf;
+impl VersionedCf for TypeAndCreationIndexedEntitiesCf {
+    type Key = (EntityType, CreationId);
+    type Value = EntityBlueprintId;
+
+    const VERSIONED_NAME: &'static str = "type_and_creation_indexed_entities";
+    type KeyCodec = TypeAndCreationIndexKeyDbCodec;
+    type VersionedValue = VersionedEntityBlueprintId;
+}
+
+/// Node IDs and blueprints of all objects, indexed by their blueprint ID and creation order.
+/// Schema: `[PackageAddress.0, hash(blueprint_name), StateVersion.to_be_bytes(), (index_within_txn as u32).to_be_bytes()].concat()` -> `scrypto_encode(VersionedObjectBlueprintName)`
+pub struct BlueprintAndCreationIndexedObjectsCf;
+impl VersionedCf for BlueprintAndCreationIndexedObjectsCf {
+    type Key = (PackageAddress, Hash, CreationId);
+    type Value = ObjectBlueprintName;
+
+    const VERSIONED_NAME: &'static str = "blueprint_and_creation_indexed_objects";
+    type KeyCodec = BlueprintAndCreationIndexKeyDbCodec;
+    type VersionedValue = VersionedObjectBlueprintName;
+}
+
+/// Substate values associated with leaf nodes of the state hash tree's Substate Tier.
+/// Needed for [`LeafSubstateValueStore`].
+/// Note: This table does not use explicit versioning wrapper, since each serialized substate
+/// value is already versioned.
+pub struct AssociatedStateTreeValuesCf;
+impl DefaultCf for AssociatedStateTreeValuesCf {
+    type Key = StoredTreeNodeKey;
+    type Value = DbSubstateValue;
+
+    const DEFAULT_NAME: &'static str = "associated_state_tree_values";
+    type KeyCodec = StoredTreeNodeKeyDbCodec;
+    type ValueCodec = DirectDbCodec;
+}
+
+/// Address book and safety state store migration status. Filled once during the migration.
+pub struct MigrationStatusCf;
+impl DefaultCf for MigrationStatusCf {
```

Comment:
So (1) these new CFs shouldn't be in `state-manager` and (2) we should make sure they're versioned, so we can update the models in future! :)

Reply:
Makes sense. Will update.